orchagent/dbggendumporch.cpp

```diff
@@ -0,0 +1,63 @@
+#include "dbggendumporch.h"
+#include "schema.h"
+
+using namespace std;
+using namespace swss;
+
+DbgGenDumpOrch::DbgGenDumpOrch(DBConnector *db, std::string table_name)
+    : Orch(db, table_name)
+{
+    SWSS_LOG_ENTER();
+}
+
+DbgGenDumpOrch::~DbgGenDumpOrch()
+{
+    SWSS_LOG_ENTER();
+}
+
+void DbgGenDumpOrch::doTask(Consumer &consumer)
+{
+    SWSS_LOG_ENTER();
+    auto it = consumer.m_toSync.begin();
+    while (it != consumer.m_toSync.end())
+    {
+        if (consumer.getTableName() == CFG_DBG_GEN_DUMP_TABLE_NAME)
+        {
+            KeyOpFieldsValuesTuple t = it->second;
+            string op = kfvOp(t);
+            if (op == SET_COMMAND)
+            {
+                auto& fieldValues = kfvFieldsValues(t);
+                auto value = fvValue(fieldValues[0]);
+                const char* value_cstr = value.c_str();
+        
+                sai_status_t ret = sai_dbg_generate_dump(value_cstr);
+                if (ret != SAI_STATUS_SUCCESS)
+                {
+                    SWSS_LOG_ERROR("Failed to generate dbg dump file %s\n", value_cstr);
+                }
+                else
+                {
+                    SWSS_LOG_DEBUG("generate dbg dump file %s successfully\n", value_cstr);
+                }
+
+                //write to config DB the return status to inform the caller
+                string ret_str = std::to_string(ret);
+                swss::DBConnector configDb("CONFIG_DB", 0); 
```

Comment:
Wouldn't it be better to connect to DB in constructor? what happens if dump generation is requested more than once?